rfcs/0002-upload-from-dropbox/text.md

```diff
@@ -387,29 +390,111 @@ will return an error.
 
 #### Processing Status and State
 
-The API documentation returns a _status_ in response. This status informs the user of the current state of the upload.
-The following diagram describes the possible state transitions:
-
-![State Diagram](https://app.lucidchart.com/publicSegments/view/9cfbd683-215a-4bb8-b3bd-f83cd709ade9/image.png)
-Figure: State Diagram
-
-There are several different states that an upload can be in. The table below defines them.
-
-| states         | Description                                                                                 |
-| -------------- | ------------------------------------------------------------------------------------------- |
-| Start          | The upload is never actually in this state.                                                 |
-| Waiting        | The upload is enqueued, and waiting for the upload container.                               |
-| Uploading      | The file is actively being uploaded.                                                        |
-| Uploaded       | The upload was completed successfully.                                                      |
-| Failed         | The upload has failed. A new link or file must be uploaded. Any upload progress is deleted. |
-| Cancel Pending | The upload is in the process of being canceled                                              |
-| Canceled       | The upload has been canceled. Any upload progress is deleted.                               |
-| Verify         | The file is being verified       |
-| Download       | The file is being downloaded from Dropbox |
-| Generating     | Files to store in S3 are generated |
-| Updating       | Updates the table with the new dataset and asset links |
-
-<a name="upload-state-table">**Table:** Upload State Table </a>
+The response to the [status endpoint](#get-datasetsdataset_idstatus) returns mutliple fields that update as processing
+progresses. When processing starts, we expect just a "Waiting" upload status:
+```
```

Comment:
I suggest nesting under the status type. It's cleaner and easier to parse.

``` json
{
  "upload": {
    "status": "Waiting",
    "message": "",
    "progress": ""
  }
}
```

Reply:
Yeah I think I can go either way. But if you want the frontend to look like

🟢  Upload
🟢  Validation
🔴  Seurat RDS
⏳  Loom
🟢  CXG

Then I think you want the more complex response. I'm pretty sure that's what we decided in the meeting, but I could be misremembering. What do you think @tihuan ?

Reply:
That's what I'd like to represent on the front end, @mckinsel.